### PR TITLE
override `getNormalLoggingLevel` to set log `Level.Fine`

### LINKED ISSUE #55: Don't continuously log messages whenever no API token is configured
**Describe the bug**
Whenever no API token is configured for the datadog plugin, the plugin will continuously write to the jenkins system log, to this affect.

As a result, the jenkins system log is in affect useless as when debugging jenkins issues, the logs have likely already been rotated. 

As an example, the plugin has written 17 log lines in 39 seconds:

```
Apr 07, 2020 8:04:56 AM INFO hudson.model.AsyncPeriodicWork lambda$doRun$0
Started Datadog Counters Publisher
Apr 07, 2020 8:04:56 AM SEVERE org.datadog.jenkins.plugins.datadog.clients.DatadogHttpClient getInstance
Datadog API Key is not set properly
Apr 07, 2020 8:04:56 AM INFO hudson.model.AsyncPeriodicWork lambda$doRun$0
Finished Datadog Counters Publisher. 5 ms
Apr 07, 2020 8:05:06 AM INFO hudson.model.AsyncPeriodicWork lambda$doRun$0
Started Datadog Counters Publisher
Apr 07, 2020 8:05:06 AM SEVERE org.datadog.jenkins.plugins.datadog.clients.DatadogHttpClient getInstance
Datadog API Key is not set properly
Apr 07, 2020 8:05:06 AM INFO hudson.model.AsyncPeriodicWork lambda$doRun$0
Finished Datadog Counters Publisher. 5 ms
Apr 07, 2020 8:05:16 AM INFO hudson.model.AsyncPeriodicWork lambda$doRun$0
Started Datadog Counters Publisher
Apr 07, 2020 8:05:16 AM SEVERE org.datadog.jenkins.plugins.datadog.clients.DatadogHttpClient getInstance
Datadog API Key is not set properly
Apr 07, 2020 8:05:16 AM INFO hudson.model.AsyncPeriodicWork lambda$doRun$0
Finished Datadog Counters Publisher. 5 ms
Apr 07, 2020 8:05:16 AM SEVERE org.datadog.jenkins.plugins.datadog.clients.DatadogHttpClient getInstance
Datadog API Key is not set properly
Apr 07, 2020 8:05:23 AM INFO hudson.model.AsyncPeriodicWork lambda$doRun$0
Started EC2 Jenkins Agents Monitor
Apr 07, 2020 8:05:23 AM INFO hudson.model.AsyncPeriodicWork lambda$doRun$0
Finished EC2 Jenkins Agents Monitor. 4 ms
Apr 07, 2020 8:05:26 AM INFO hudson.model.AsyncPeriodicWork lambda$doRun$0
Started Datadog Counters Publisher
Apr 07, 2020 8:05:26 AM SEVERE org.datadog.jenkins.plugins.datadog.clients.DatadogHttpClient getInstance
Datadog API Key is not set properly
Apr 07, 2020 8:05:26 AM INFO hudson.model.AsyncPeriodicWork lambda$doRun$0
Finished Datadog Counters Publisher. 5 ms
Apr 07, 2020 8:05:36 AM INFO hudson.model.AsyncPeriodicWork lambda$doRun$0
Started Datadog Counters Publisher
Apr 07, 2020 8:05:36 AM SEVERE org.datadog.jenkins.plugins.datadog.clients.DatadogHttpClient getInstance
Datadog API Key is not set properly
```
FYI, this is not the same issue reported here - while they are both related in the same area, the below was a regression which meant the logs were firing as exceptions:
 - https://github.com/jenkinsci/datadog-plugin/issues/42

**To Reproduce**
1. Install latest datadog plugin (though this doesn't look to be a regression, it's been the behaviour for some time)
2. Make sure an API token is not configured for the plugin (or the current one has expired)
3. observe the Jenkins log

**Expected behavior**
The datadog plugin should not be spamming the jenkins log in such a way, that it makes it un-usable for debugging other issues which may occur. 

Other plugins, such as the github plugin, do write to the log when an API token is not valid, e.g when it fails to create webhooks, but not in such high frequency.

**Additional context**
We operate an environment where we provide jenkins masters to other teams, maintaining a core set of plugins. Not all teams will leverage the plugins we have installed, such as the datadog plugin, and may not configure an API token for the plugin. 

Others may have expired API tokens and as a result, if we need to debug any issues, the logs are not usable.

### PR DESCRIPTION
**Pull request by jetersen**
### Requirements for Contributing to this repository

* Fill out the template below. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* The pull request must only fix one issue at the time.
* The pull request must update the test suite to demonstrate the changed functionality.
* After you create the pull request, all status checks must be pass before a maintainer reviews your contribution. For more details, please see [CONTRIBUTING](/CONTRIBUTING.md). 

### What does this PR do?

Reduce logging noise
fixes #55 

### Description of the Change

Overridden the `getNormalLoggingLevel` to set log level to fine

### Alternate Designs

N/A

### Possible Drawbacks

None

### Verification Process

Less noisy logs

### Additional Notes

None

### Release Notes

Fix noisy logs

### Review checklist (to be filled by reviewers)

- [ ] Feature or bug fix MUST have appropriate tests (unit, integration, etc...)
- [ ] PR title must be written as a CHANGELOG entry [(see why)](/CONTRIBUTING.md#pull-request-title)
- [ ] Files changes must correspond to the primary purpose of the PR as described in the title (small unrelated changes should have their own PR)
- [ ] PR must have one `changelog/` label attached. If applicable it should have the `backward-incompatible` label attached.
- [ ] PR should not have `do-not-merge/` label attached.
- [ ] If Applicable, issue must have `kind/` and `severity/` labels attached at least.

### CODE REVIEW

File: src/main/java/org/datadog/jenkins/plugins/datadog/clients/ClientFactory.java [hunk 1]
```diff
@@ -62,6 +62,9 @@ public static DatadogClient getClient() {
             targetPort = descriptor.getTargetPort();
             targetLogCollectionPort = descriptor.getTargetLogCollectionPort();
         }
+        if (targetApiKey == null) {
```
Comment (jetersen) [inline src/main/java/org/datadog/jenkins/plugins/datadog/clients/ClientFactory.java]: Without a valid API key it does not make sense to return a client.
However, I now see this affects many callers, I will reduce the area to async period worker

**Comment by sarah-witt** [comment]
Hi @jetersen! Thank you for your contribution! The logging level of `DatadogCountersPublisher` is a great addition and something we should keep and merge to address #55, but the approach of checking for a null API key in `DatadogCountersPublishers` has a few drawbacks. With this approach, there will never be a log line about a missing API key, which is something that we _do_ want (but not a continuous stream of them!). It also does not address the cases in which a different required field is not set properly, such as a URL, which would then cause a continuous stream of different error messages. We encountered these same difficulties when looking into the issue, which is why we decided to do the refactoring in #59 to account for these cases. Your addition of overriding `getNormalLoggingLevel()` addresses one source of the continuous logs, but what do you think about letting #59 fix the other source (null API key)?

**Comment by sarah-witt** [comment]
Yes, #59 is meant to fix the issue where there are logs about a missing API key every 10 seconds. If you see a drawback to the approach, or something that we missed, please let us know!

File: src/main/java/org/datadog/jenkins/plugins/datadog/publishers/DatadogCountersPublisher.java [hunk 2]
```diff
@@ -51,6 +53,10 @@ public long getRecurrencePeriod() {
 
     @Override
     protected void execute(TaskListener taskListener) throws IOException, InterruptedException {
+        DatadogGlobalConfiguration datadogGlobalDescriptor = DatadogUtilities.getDatadogGlobalDescriptor();
+        if (datadogGlobalDescriptor != null && datadogGlobalDescriptor.getTargetApiKey() == null) {
+            return;
+        }
```
Comment (jetersen) [inline src/main/java/org/datadog/jenkins/plugins/datadog/publishers/DatadogCountersPublisher.java:59]:
```suggestion
```

File: src/main/java/org/datadog/jenkins/plugins/datadog/publishers/DatadogCountersPublisher.java [hunk 3]
```diff
@@ -27,7 +27,9 @@ of this software and associated documentation files (the "Software"), to deal
 
 import hudson.Extension;
 import hudson.model.*;
+import java.util.logging.Level;
 import org.datadog.jenkins.plugins.datadog.DatadogClient;
+import org.datadog.jenkins.plugins.datadog.DatadogGlobalConfiguration;
```
Comment (jetersen) [inline src/main/java/org/datadog/jenkins/plugins/datadog/publishers/DatadogCountersPublisher.java:32]:
```suggestion
```

**Comment by jetersen** [comment]
Feel free to squash and merge @sarah-witt

**Pull request merged**